src/App.tsx

```diff
@@ -3,83 +3,85 @@ import 'bulma/css/bulma.css';
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
+  { name: 'Dumplings' },
+  { name: 'Carrot' },
+  { name: 'Eggs' },
+  { name: 'Ice cream' },
+  { name: 'Apple' },
+  { name: 'Bread' },
+  { name: 'Fish' },
+  { name: 'Honey' },
+  { name: 'Jam' },
+  { name: 'Garlic' },
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+type Props = {};
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+export class App extends React.Component<Props, State> {
+  state: Readonly<State> = {
+    // eslint-disable-next-line react/no-unused-state
+    selectedGood: '',
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render(): React.ReactNode {
+    return (
+      <main className="section container">
+        <h1 className="title is-flex is-align-items-center">
+          {this.state.selectedGood
+            ? `${this.state.selectedGood} is selected`
+            : 'No goods selected'}
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => this.setState({ selectedGood: '' })}
+          />
+        </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map((item) => (
+              <tr
+                data-cy="Good"
+                key={item.name}
+                className={`${
+                  item.name === this.state.selectedGood
+                    ? 'has-background-success-light'
+                    : ''
+                }`}
+              >
+                <td>
+                  <button
+                    data-cy="AddButton"
+                    type="button"
+                    className={`button ${
+                      item.name === this.state.selectedGood ? 'is-info' : ''
+                    }`}
+                    onClick={() => {
+                      if (this.state.selectedGood === item.name) {
+                        this.setState({ selectedGood: '' });
+                      } else {
+                        this.setState({ selectedGood: item.name });
+                      }
+                    }}
+                  >
+                    {item.name === this.state.selectedGood ? '-' : '+'}
```

Comment:
you repeat this check many times, move it to separate function or create separate boolean variable for it